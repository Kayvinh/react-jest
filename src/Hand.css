.Hand {
  border: solid 2px white;
  border-radius: 20px;
  margin: 20px;
  padding: 10px;
  width: 630px;
  height: 300px;
}