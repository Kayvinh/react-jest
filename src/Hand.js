import React from "react";
import Card from "./Card";
import "./Hand.css";

/** Show a hand of cards.
 *
 * Props:
 * - cards: [ { rank, suit }, ... ]
 */

function Hand({cards}) {
  return (
    <div className="Hand">
      {cards.map((c, idx) =>
          <Card
            key={idx}
              rank={c.rank}
              suit={c.suit}/>)}
    </div>
  );
}

// end

export default Hand;
