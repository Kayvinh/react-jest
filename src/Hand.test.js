import React from "react";
import { render, screen } from "@testing-library/react";
import Hand from "./Hand";

const hand = [{ rank: "7", suit: "H" }, { rank: "A", suit: "S" }];

it("renders 2 cards", function () {
  render(<Hand cards={hand} />);
  const cardElements = screen.getAllByRole("img", { name: /./, class: "Card" });
    expect(cardElements).toHaveLength(hand.length);
});

it("matches snapshot", function () {
  const { container } = render(<Hand cards={hand} />);
  expect(container).toMatchSnapshot();
});
