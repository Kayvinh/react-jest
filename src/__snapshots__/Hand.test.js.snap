// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Hand"
  >
    <img
      alt="7H"
      class="Card"
      src="https://deckofcardsapi.com/static/img/7H.png"
    />
    <img
      alt="AS"
      class="Card"
      src="https://deckofcardsapi.com/static/img/AS.png"
    />
  </div>
</div>
`;
